kobo-uncaged/main.go

```diff
@@ -315,67 +310,9 @@ func (ku *KoboUncaged) getKoboInfo() error {
 		versFields := strings.Split(vers, ",")
 		fwStr := strings.Split(versFields[2], ".")
 		for i, f := range fwStr {
-			ku.koboInfo.fw[i], _ = strconv.Atoi(f)
+			ku.fw[i], _ = strconv.Atoi(f)
 		}
-		ku.koboInfo.model = koboDeviceID(versFields[len(versFields)-1])
-	}
-	// Once we have the model number, we set the appropriate cover image dims
-	// These values come from https://github.com/kovidgoyal/calibre/blob/master/src/calibre/devices/kobo/driver.py
-	switch ku.koboInfo.model {
-	case glo, aura, auraEd2r1, auraEd2r2:
-		ku.koboInfo.coverDetails[fullCover] = coverDims{width: 758, height: 1024}
-		ku.koboInfo.coverDetails[libFull] = coverDims{width: 355, height: 479}
-		ku.koboInfo.coverDetails[libGrid] = coverDims{width: 149, height: 201}
-	case gloHD, claraHD:
-		ku.koboInfo.coverDetails[fullCover] = coverDims{width: 1072, height: 1448}
-		ku.koboInfo.coverDetails[libFull] = coverDims{width: 355, height: 479}
-		ku.koboInfo.coverDetails[libGrid] = coverDims{width: 149, height: 201}
-	case auraHD, auraH2O, auraH2Oed2r1, auraH2Oed2r2:
-		ku.koboInfo.coverDetails[fullCover] = coverDims{width: 1080, height: 1440}
-		ku.koboInfo.coverDetails[libFull] = coverDims{width: 355, height: 471}
-		ku.koboInfo.coverDetails[libGrid] = coverDims{width: 149, height: 198}
-	case auraOne, auraOneLE:
-		ku.koboInfo.coverDetails[fullCover] = coverDims{width: 1404, height: 1872}
-		ku.koboInfo.coverDetails[libFull] = coverDims{width: 355, height: 473}
-		ku.koboInfo.coverDetails[libGrid] = coverDims{width: 149, height: 198}
-	case forma:
-		ku.koboInfo.coverDetails[fullCover] = coverDims{width: 1440, height: 1920}
-		ku.koboInfo.coverDetails[libFull] = coverDims{width: 355, height: 473}
-		ku.koboInfo.coverDetails[libGrid] = coverDims{width: 149, height: 198}
-	default:
-		ku.koboInfo.coverDetails[fullCover] = coverDims{width: 600, height: 800}
-		ku.koboInfo.coverDetails[libFull] = coverDims{width: 355, height: 473}
-		ku.koboInfo.coverDetails[libGrid] = coverDims{width: 149, height: 198}
-	}
-
-	// Populate model name
-	switch ku.koboInfo.model {
-	case touch2, touchAB, touchC:
-		ku.koboInfo.modelName = "Touch"
-	case mini:
-		ku.koboInfo.modelName = "Mini"
-	case glo:
-		ku.koboInfo.modelName = "Glo"
-	case gloHD:
-		ku.koboInfo.modelName = "Glo HD"
-	case aura:
-		ku.koboInfo.modelName = "Aura"
-	case auraH2O:
-		ku.koboInfo.modelName = "Aura H2O"
-	case auraH2Oed2r1, auraH2Oed2r2:
-		ku.koboInfo.modelName = "Aura H2O Ed. 2"
-	case auraEd2r1, auraEd2r2:
-		ku.koboInfo.modelName = "Aura Ed. 2"
-	case auraHD:
-		ku.koboInfo.modelName = "Aura HD"
-	case auraOne, auraOneLE:
-		ku.koboInfo.modelName = "Aura One"
-	case claraHD:
-		ku.koboInfo.modelName = "Clara HD"
-	case forma:
-		ku.koboInfo.modelName = "Forma"
-	default:
-		ku.koboInfo.modelName = "Unknown Kobo"
+		ku.device = koboDevice(versFields[len(versFields)-1])
```

Comment:
Do you think it might be a good idea to ensure the device actually is known here? The benefit to this is that errors parsing the file would be more visible, but the downside would be that it would be less future-proof. An alternative could be to just check the format of the device id.

Reply:
Hmm, tough call. Knowing the model is a nice-to-have, but probably not critical. The only real purpose is to have appropriate sized thumbnails, but there is a fallback to that as well.